lib/type/string.ex

```diff
@@ -20,4 +20,74 @@ defmodule Litmus.Type.String do
           trim: boolean,
           required: boolean
         }
+
+  @spec validate_field(t, binary, map) :: {:ok, map} | {:error, binary}
+  def validate_field(type, field, data) do
+    with {:ok, data} <- Required.validate(type, field, data),
+         {:ok, data} <- min_length_validate(type, field, data),
+         {:ok, data} <- max_length_validate(type, field, data),
+         {:ok, data} <- length_validate(type, field, data) do
+      {:ok, data}
+    else
+      {:error, msg} -> {:error, msg}
+    end
+  end
+
+  @spec min_length_validate(t, binary, map) :: {:ok, map} | {:error, binary}
+  def min_length_validate(%Litmus.Type.String{min_length: min_length}, field, params)
+      when min_length != nil and is_integer(min_length) and min_length >= 0 do
+    if Map.has_key?(params, field) do
+      if String.length(params[field]) >= min_length do
+        {:ok, params}
+      else
+        {:error, "#{field} length must be more than or equal to #{min_length} characters long"}
+      end
+    else
+      {:ok, params}
+    end
+  end
+
+  def min_length_validate(%Litmus.Type.String{min_length: nil}, _field, params),
+    do: {:ok, params}
+
+  @spec max_length_validate(t, binary, map) :: {:ok, map} | {:error, binary}
+  def max_length_validate(%Litmus.Type.String{max_length: max_length}, field, params)
+      when max_length != nil and is_integer(max_length) and max_length >= 0 do
+    if Map.has_key?(params, field) do
```

Comment:
Instead of having three cases, you can just do 
```elixir
if Map.has_key?(params, field) && String.length(params[field]) > max_length do
  {:error, "#{field} length must be less than or equal to #{max_length} characters long"}
else
  {:ok, params}
end
```

Reply:
can we pattern match the field in the params in the function header?

```elixir
def max_length_validate(%Litmus.Type.String{max_length: max_length}, field, %{field => value})

end
```

Reply:
Thought of doing this before. The problem is, that if the schema and data are:
```
  type = %Type.String{
      max_length: 3
    }

    field = :id

    data = %{
    }
```

The validate function should return `{:ok, data}` , since the field is not required.

Reply:
sorry! @kyle your solution would work too

Reply:
@mgartner , didnt know pattern matching worked that way. will try it out